fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/DistinctSplit.java

```diff
@@ -0,0 +1,282 @@
+        List<Alias> otherAggFuncs = new ArrayList<>();
+        if (!needTransform((LogicalAggregate<Plan>) agg, distinctFuncWithAlias, otherAggFuncs)) {
+            return agg;
+        }
+
+        LogicalAggregate<Plan> cloneAgg = (LogicalAggregate<Plan>) LogicalPlanDeepCopier.INSTANCE
+                .deepCopy(agg, new DeepCopierContext());
+        LogicalCTEProducer<Plan> producer = new LogicalCTEProducer<>(ctx.statementContext.getNextCTEId(),
+                cloneAgg.child());
+        ctx.cteProducerList.add(producer);
+        Map<Slot, Slot> originToProducerSlot = new HashMap<>();
+        for (int i = 0; i < agg.child().getOutput().size(); ++i) {
+            Slot originSlot = agg.child().getOutput().get(i);
+            Slot cloneSlot = cloneAgg.child().getOutput().get(i);
+            originToProducerSlot.put(originSlot, cloneSlot);
+        }
+        distinctFuncWithAlias = ExpressionUtils.replace((List) distinctFuncWithAlias, originToProducerSlot);
+        otherAggFuncs = ExpressionUtils.replace((List) otherAggFuncs, originToProducerSlot);
+        // construct cte consumer and aggregate
+        List<LogicalAggregate<Plan>> newAggs = new ArrayList<>();
+        // All otherAggFuncs are placed in the first one
+        Map<Alias, Alias> newToOriginDistinctFuncAlias = new HashMap<>();
+        List<Expression> outputJoinGroupBys = new ArrayList<>();
+        for (int i = 0; i < distinctFuncWithAlias.size(); ++i) {
+            Expression distinctAggFunc = distinctFuncWithAlias.get(i).child(0);
+            LogicalCTEConsumer consumer = new LogicalCTEConsumer(ctx.statementContext.getNextRelationId(),
+                    producer.getCteId(), "", producer);
+            ctx.cascadesContext.putCTEIdToConsumer(consumer);
+            Map<Slot, Slot> producerToConsumerSlotMap = new HashMap<>();
+            for (Map.Entry<Slot, Slot> entry : consumer.getConsumerToProducerOutputMap().entrySet()) {
+                producerToConsumerSlotMap.put(entry.getValue(), entry.getKey());
+            }
+            List<Expression> replacedGroupBy = ExpressionUtils.replace(cloneAgg.getGroupByExpressions(),
+                    producerToConsumerSlotMap);
+            Expression newDistinctAggFunc = ExpressionUtils.replace(distinctAggFunc, producerToConsumerSlotMap);
+            List<NamedExpression> outputExpressions = replacedGroupBy.stream()
+                    .map(Slot.class::cast).collect(Collectors.toList());
+            Alias alias = new Alias(newDistinctAggFunc);
+            outputExpressions.add(alias);
+            if (i == 0) {
+                // save replacedGroupBy
+                outputJoinGroupBys.addAll(replacedGroupBy);
+            }
+            LogicalAggregate<Plan> newAgg = new LogicalAggregate<>(replacedGroupBy, outputExpressions, consumer);
+            newAggs.add(newAgg);
+            newToOriginDistinctFuncAlias.put(alias, distinctFuncWithAlias.get(i));
```

Comment:
It seems like some duplicate code exists between non-distinct-aggregte and distinct-aggregate, we should reuse it